scalafix-interfaces/src/main/java/scalafix/interfaces/Scalafix.java

```diff
@@ -58,15 +67,96 @@
+     * classload an instance of it via runtime reflection.
+     * <p>
+     * The custom classloader optionally provided with {@link ScalafixArguments#withToolClasspath} to compile and
+     * classload external rules must have the classloader of the returned instance as ancestor to share a common
+     * loaded instance of `scalafix-core`, and therefore have been compiled against the requested Scala binary version.
+     *
+     * @param scalaBinaryVersion The Scala binary version ("2.13" for example) available in the classloader of the
+     *                           returned instance. To be able to run advanced semantic rules using the Scala
+     *                           Presentation Compiler such as ExplicitResultTypes, this must match the binary
+     *                           version that the target classpath was built with, as provided with
+     *                           {@link ScalafixArguments#withScalaVersion}.
+     * @param repositories       Maven/Ivy repositories to fetch the JARs from.
+     * @return An implementation of the {@link Scalafix} interface.
+     * @throws ScalafixException in case of errors during artifact resolution/fetching.
+     */
+    static Scalafix fetchAndClassloadInstance(String scalaBinaryVersion, List<Repository> repositories)
+            throws ScalafixException {
+
+        String scalaVersionKey;
+        switch (scalaBinaryVersion) {
+            case "2.11":
+                scalaVersionKey = "scala211";
+                break;
+            case "2.12":
+                scalaVersionKey = "scala212";
+                break;
+            case "2.13":
+                scalaVersionKey = "scala213";
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported scala version " + scalaBinaryVersion);
+        }
 
+        Properties properties = new Properties();
+        String propertiesPath = "scalafix-interfaces.properties";
+        InputStream stream = Scalafix.class.getClassLoader().getResourceAsStream(propertiesPath);
+        try {
+            properties.load(stream);
+        } catch (IOException | NullPointerException e) {
+            throw new ScalafixException("Failed to load '" + propertiesPath + "' to lookup versions", e);
+        }
+
+        String scalafixVersion = properties.getProperty("scalafixVersion");
+        String scalaVersion = properties.getProperty(scalaVersionKey);
+        if (scalafixVersion == null || scalaVersion == null)
+            throw new ScalafixException("Failed to lookup versions from '" + propertiesPath + "'");
+
+        List<URL> jars = ScalafixCoursier.scalafixCliJars(repositories, scalafixVersion, scalaVersion);
+        ClassLoader parent = new ScalafixInterfacesClassloader(Scalafix.class.getClassLoader());
+        return classloadInstance(new URLClassLoader(jars.stream().toArray(URL[]::new), parent));
```

Comment:
can we assume java8 at runtime since that's what CI run tests with?

Reply:
Correct. It’s ok to target java 8